### ISSUE TITLE
Question about building process

### ISSUE THREAD

**Issue by werdahias** [jellyfin/jellyfin-desktop#390]
I'm considering taking a stab at an official debian package. Looking at the source, I found two issues so far:
1) the dependencies under the external folder: qhttp and qslog would need packaging from scratch. I take it HIDremote is also needed for a linux-only build ? The other ones look like mac-only to me.
2) During the build process, the download_webclient.sh script is called. Could you maybe elaborate on this ? Is this strictly needed for the build?
3) Is linking against the libmpv from the debian archive supported ?

**Comment by iwalton3** [comment]
> 1) the dependencies under the external folder: qhttp and qslog would need packaging from scratch. I take it HIDremote is also needed for a linux-only build? The other ones look like mac-only to me.

HIDremote looks like it is mac only. https://github.com/jellyfin/jellyfin-media-player/blob/9a08120166993b69f0cea4f40a6a7903df4b8059/src/CMakeLists.txt#L84

The qhttp dependency may be dead code. Plex Media Player, which is what this project inherited code from, used to run local HTTP server for control by the mobile app. This application doesn't do that. I'm not sure about qslog offhand.

> 2) During the build process, the download_webclient.sh script is called. Could you maybe elaborate on this ? Is this strictly needed for the build?

Yes. It downloads the web client here which is vital to the operation of the application: https://github.com/iwalton3/jellyfin-web-jmp/releases/tag/jwc-10.8.9

It would likely be packaged as a separate package for a "proper" Debian package. I use the `download_webclient.sh` as a shortcut to reduce the amount of files I have to update when doing a release.

If this is a significant issue for you I can move or adjust that functionality in the CI job so it doesn't interfere with your efforts to make a package.

> 3) Is linking against the libmpv from the debian archive supported?

Yes. The Debian packages we distribute use the packaged libmpv from upstream. The one in Debian Stable is known to work.

**Comment by werdahias** [comment]
Thanks for the quick reply :) So if I understand this correctly, the web [client](https://github.com/jellyfin/jellyfin-web)  needs to be packaged first, then the player. The server is independent, and could be distributed seperately.

**Comment by iwalton3** [comment]
Correct. The web client this depends on is just a standalone build.

**Comment by werdahias** [comment]
Ok, thanks. I will look into this and tackle the web client first.

**Comment by werdahias** [comment]
I had a look at the web client yesterday and it has quite  a few missing dependencies. I will track the status on the Debian bug report page.

**Comment by iwalton3** [comment]
Hmm packaging the web app may be infeasible because of the nodejs dependency mess that prevents a lot of modern web apps being packaged.

I *might* be able to sidestep this by creating some basic UI that allows you to specify a Jellyfin server and then at that point it would just use the web client provided by the user. I wouldn't make it the default though, just something that can be enabled via command line (or the state of not having the web app packed with the app.)

Let me know if you are interested in this.

**Comment by iwalton3** [comment]
This mode of operation might also be better for an upstream Debian package anyways since it is probably less likely to break if Jellyfin Media Player's version is substantially different from the Jellyfin server.

The web client and the server are more tightly coupled version wise than the web client and my native playback API.

**Comment by werdahias** [comment]
That would indeed simplify things as I wouldn't have to package the webclient. While I have some experience with packaging, JS is new terrain for me. Maybe it's just not feasible to ship the server/web client.

**Comment by werdahias** [comment]
indeed: https://gist.github.com/iwalton3/fe29f891d6c9993032d5d57f4917d6f5?permalink_comment_id=4496805#gistcomment-4496805

**Comment by iwalton3** [comment]
Wow that was a lot more effort than I thought it would be... but you can now run the app without the bundled web client!

https://github.com/jellyfin/jellyfin-media-player/commit/2450ae1956284ebaee5ee1ebe9f207722b041dc9

**Comment by werdahias** [comment]
awesome, thanks. I'll start on the packaging process if I can make some time

**Comment by sevenrats** [comment]
what is the status of this issue?
